### PR TITLE
Updated ulimit warning message to include the recommended min value.

### PR DESCRIPTION
**Pull request by jpoehls**
Also moved the minimum ulimit value into a constant and the whole ulimit check into its own function.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -115,6 +106,23 @@ func main() {
...
+	const min = 4096
+
+	// Warn if ulimit is too low for production sites
+	if runtime.GOOS == "linux" || runtime.GOOS == "darwin" {
+		out, err := exec.Command("sh", "-c", "ulimit -n").Output() // use sh because ulimit isn't in Linux $PATH
+		if err == nil {
+			// Note that an error here need not be reported
+			lim, err := strconv.Atoi(string(bytes.TrimSpace(out)))
+			if err == nil && lim < min {
+				fmt.Printf("Warning: File descriptor limit %d is too low for production sites. Recommend at least ulimit -n %d\n", lim, min)
```
Comment (abiosoft) [inline main.go]: What do think about putting the second sentence on new line and rephrase as
```
At least %d is recommended. Set with "ulimit -n %d".
```

**Comment by mholt** [comment]
Big win - thanks!

File: main.go (hunk 1 above)
Comment (mholt) [inline main.go]: Oops. Race condition between us. :smile:

Comment (abiosoft) [inline main.go]: :) This is something minor so it's okay.

Comment (jpoehls) [inline main.go]: I like the second line idea. Can I still update this PR or do I need to open a new one for that tweak?

Comment (mholt) [inline main.go]: I think you can update the branch you made and then issue a new PR.

**Pull request merged**